go/ai/prompt.go

```diff
@@ -496,9 +491,24 @@ func loadPromptDir(r *registry.Registry, dir string, namespace string) error {
 	return nil
 }
 
-// definePartial registers a partial template in the registry.
-func definePartial(r *registry.Registry, name string, source string) {
-	// TODO: Add this functionality
+// DefinePartial adds the partial to the list of partials to the dotprompt instance
+func DefinePartial(r *registry.Registry, name string, source string) {
+	if r.Dotprompt == nil {
```

Comment:
I would say this logic should live in the registry and this function should call `registry.RegisterPartial` and `registry.RegisterHelper`.